### ISSUE TITLE
[i2c] Wakeup from sleep using pinmux

### ISSUE THREAD

**Issue by GregAC** [lowRISC/opentitan#17354]
It is a requirement that earlgrey is able to wakeup from sleep on an incoming I2C transaction (though it's permissible for that transaction to get dropped), I believe this will be implemented via the wakeup detectors in the pinmux. These detectors run on the aon clock, at 200 kHz. For i2c fast-mode and fast-mode plus the SCL clock runs at (up to) 400 kHz and 1 MHz respectively so the SDA and SCL edges will be too rapid to correctly detect (though given enough activity the wakeup logic would likely get lucky and see a edge at its slow sample rate). For standard-mode SCL runs at (up to) 100 kHz, which should be slow enough for the wakeup logic to correctly detect edges (still a slower SCL would be preferable allowing the wakeup logic to get several samples per SCL clock period rather than 2).

We need to confirm this wakeup behaviour is sufficient for requirements.

**Comment by johngt** [comment]
@moidx - do you mind confirming for I2C if wake-up behaviour detailed here is sufficient?

**Comment by a-will** [comment]
If it's only needed in normal sleep, then the "peri" clock group should still be operating, right? _If_ that is the case, this simplifies significantly, and there is no need for additional hardware in the AON domain.

**Comment by a-pronin** [comment]
> If that is the case

yes, for **our** use case, deep sleep can only happen when AP is in S3 or lower state, so there's no need to watch for activity on i2c bus connected to AP. There's a separate platform reset signal that will wake OT on resume from S3-or-lower.

**Comment by marnovandermaas** [comment]
I am changing the priority to P3 because even though the wakeup messages use a higher frequency, they will be retried if there is no response.

**Comment by GregAC** [comment]
> If it's only needed in normal sleep, then the "peri" clock group should still be operating, right? _If_ that is the case, this simplifies significantly, and there is no need for additional hardware in the AON domain.

Going by the documentation (https://opentitan.org/book/hw/ip/pwrmgr/doc/theory_of_operation.html#normal-sleep) the peri group isn't active, only the aon (200 kHz) domain is active.

> so, we need to wake up on the 1st transaction in all supported modes.

With the current hardware we're relying on getting lucky at when sampling the transaction at 200 kHz seeing an edge. Chances are this will happen especially if the transaction retries.

**Comment by a-will** [comment]
@GregAC Thanks for the correction! I apparently confused wfi clock gating with a sleep mode. The `clkmgr` docs provided info about the mechanism and a justification for leaving the I/O clocks on, but... I also missed that it later said the feature isn't currently supported.

Sorry for creating a false hope, haha. So we have normal sleep, deep sleep, and wfi (sometimes described as "processor sleep").

**Comment by msfschaffner** [comment]
Given that we've decided that it 200kHz should be fast enough for I2C wakeup in the I2C clarification session last week, I am hence closing this issue (but please reopen if you think it has not been fully addressed).
